nni/retiarii/execution/base.py

```diff
@@ -10,20 +10,31 @@
 from .interface import AbstractExecutionEngine, AbstractGraphListener
 from .. import codegen, utils
 from ..graph import Model, ModelStatus, MetricData, Evaluator
-from ..integration_api import send_trial, receive_trial_parameters, get_advisor
+from ..integration_api import send_trial, receive_trial_parameters, get_advisor, report_search_space
 
 _logger = logging.getLogger(__name__)
 
+def extract_visual_hp(model: Model) -> dict:
+    hp_config = {}
+    for mut in model.history:
+        if len(mut.samples) == 1:
+            hp_config[mut.mutator.label] = mut.samples[0]
+        else:
+            for i, sample in enumerate(mut.samples):
+                hp_config[f'{mut.mutator.label}_{i}'] = sample
+    return hp_config
 
 class BaseGraphData:
-    def __init__(self, model_script: str, evaluator: Evaluator) -> None:
+    def __init__(self, model_script: str, evaluator: Evaluator, visual_hp: dict = None) -> None:
```

Comment:
Suggest renaming it into something like `mutation_summary`. It's not for visualization purpose only.

Reply:
good point